Simplified/NYPLMyBooksDownloadCenter.m

```diff
@@ -774,6 +799,36 @@ - (void)startDownloadForBook:(NYPLBook *const)book
   }
 }
 
+- (void)addDownloadTaskWithRequest:(NSURLRequest *)request
+                              book:(NYPLBook *)book {
+  NSURLSessionDownloadTask *const task = [self.session downloadTaskWithRequest:request];
+  
+  self.bookIdentifierToDownloadInfo[book.identifier] =
+    [[NYPLMyBooksDownloadInfo alloc]
+     initWithDownloadProgress:0.0
+     downloadTask:task
+     rightsManagement:NYPLMyBooksDownloadRightsManagementUnknown];
+  
+  self.taskIdentifierToBook[@(task.taskIdentifier)] = book;
```

Comment:
wondering if we should have a safety guard here to check if `book` is not nil to prevent a possible crash.